File: sample-objc/SendBird-iOS/GroupChannel/ViewCell/CreateGroupChannelUserListTableViewCell.m
//
//  CreateGroupChannelUserListTableViewCell.m
//  SendBird-iOS
//
//  Created by Jed Kyung on 9/21/16.
//  Copyright © 2016 SendBird. All rights reserved.
//

#import <AFNetworking/UIImageView+AFNetworking.h>
#import "CreateGroupChannelUserListTableViewCell.h"

@interface CreateGroupChannelUserListTableViewCell()

@property (weak, nonatomic) IBOutlet UIImageView *profileImageView;
@property (weak, nonatomic) IBOutlet UILabel *nicknameLabel;
@property (weak, nonatomic) IBOutlet UIImageView *checkImageView;

@property (strong, nonatomic) SBDUser *user;

@end

@implementation CreateGroupChannelUserListTableViewCell

+ (UINib *)nib
{
    return [UINib nibWithNibName:NSStringFromClass([self class]) bundle:[NSBundle bundleForClass:[self class]]];
}

+ (NSString *)cellReuseIdentifier
{
    return NSStringFromClass([self class]);
}

- (void)setModel:(SBDUser *)aUser {
    self.user = aUser;
    [self.profileImageView setImageWithURL:[NSURL URLWithString:self.user.profileUrl] placeholderImage:[UIImage imageNamed:@"img_profile"]];
    self.nicknameLabel.text = self.user.nickname;
}

- (void)setSelectedUser:(BOOL)selected {
    if (selected) {
        [self.checkImageView setImage:[UIImage imageNamed:@"btn_check_on"]];
    }
    else {
        [self.checkImageView setImage:[UIImage imageNamed:@"btn_check_off"]];
    }
}

@end

File: sample-swift/SendBird-iOS/GroupChannel/ViewCell/CreateGroupChannelUserListTableViewCell.swift
//
//  CreateGroupChannelUserListTableViewCell.swift
//  SendBird-iOS
//
//  Created by Jed Kyung on 10/9/16.
//  Copyright © 2016 SendBird. All rights reserved.
//

import UIKit
import SendBirdSDK
import AlamofireImage

class CreateGroupChannelUserListTableViewCell: UITableViewCell {
    @IBOutlet weak var profileImageView: UIImageView!
    @IBOutlet weak var nicknameLabel: UILabel!
    @IBOutlet weak var checkImageView: UIImageView!
    
    private var user: SBDUser!

    static func nib() -> UINib {
        return UINib(nibName: String(describing: self), bundle: Bundle(for: self))
    }
    
    static func cellReuseIdentifier() -> String {
        return String(describing: self)
    }
    
    func setModel(aUser: SBDUser) {
        self.user = aUser
        self.profileImageView.af_setImage(withURL: URL(string: self.user.profileUrl!)!, placeholderImage:UIImage(named: "img_profile"))
        self.nicknameLabel.text = self.user.nickname
    }
    
    func setSelectedUser(selected: Bool) {
        if selected {
            self.checkImageView.image = UIImage(named: "btn_check_on")
        }
        else {
            self.checkImageView.image = UIImage(named: "btn_check_off")
        }
    }
}
